syntax = "proto3";

package customer;

service CustomerService {
  rpc GetCustomer(GetCustomerRequest) returns (CustomerResponse);
  rpc UpdateCustomer(UpdateCustomerRequest) returns (CustomerResponse);
}

message GetCustomerRequest {
  string customer_id = 1;
}

message UpdateCustomerRequest {
  string customer_id = 1;
  string name = 2;
  string email = 3;
  string phone = 4;
  repeated string delivery_addresses = 5;
}

message CustomerResponse {
  string customer_id = 1;
  string name = 2;
  string email = 3;
  string phone = 4;
  repeated string delivery_addresses = 5;
}